src/views/EditUser.js

```diff
@@ -16,28 +18,80 @@ const useStyles = makeStyles(() => ({
     },
     add_button: {
         color: '#04bdaf'
+    },
+    user_info_container: {
+        display: 'grid',
+        gridTemplateColumns: '40% auto',
+        gridGap: '15px',
+        width: '100%',
+        padding: '10px',
+        height: '100%'
+    },
+    user_info: {
+        display: 'flex',
+        flexDirection: 'column',
+        padding: '0px 20px 20px 20px',
+        maxHeight: '100%',
+        overflowY: 'auto'
+    },
+    margin: {
+        margin: theme.spacing(1),
     }
 }));
 
-export const EditUser = (props) => {
-    // navigate to users page if no user info is passed along
-    if (!props.location.state) navigate('/users')
+export const EditUser = () => {
     // navigate to login page if no token is found
     if (!localStorage.getItem('zmt-token')) navigate('/')
-    
     const auth = localStorage.getItem('zmt-token');
-    const currentUser = props.location.state ? props.location.state.userInfo : new Array(2);
+    const loggedUser = localStorage.getItem('zmt-username')
+    const params = new URLSearchParams(location.search)
+    const currentUser = params.get('username')
     const classes = useStyles();
     const [isLoading, setLoading] = useState(false);
     const [refresh, setRefresh] = useState(false);
     const { restApiLocation } = useEnvironment();
-    const { zoneName } = useServer();
+    const { zoneName, loadUser } = useServer();
     const [groupsOfUser, setGroupOfUser] = useState([]);
     const [filterGroupName, setFilterName] = useState('');
     const [filterGroupNameResult, setFilterNameResult] = useState();
+    const [userType, setUserType] = useState({
+        value: '',
+        status: ''
+    })
+    const [password, setPassword] = useState({
+        newPassword: '',
+        confirmPassword: '',
+        showPassword: false,
+        showConfirmPassword: false,
+        status: ''
+    })
+
+    useEffect(() => {
+        // hide this interface for the current rodsadmin user
+        if (loggedUser === currentUser) navigate('/users')
+        else {
+            axios({
+                method: 'GET',
+                url: `${restApiLocation}/query`,
+                headers: {
+                    'Accept': 'application/json',
+                    'Authorization': auth
+                },
+                params: {
+                    query_string: `SELECT USER_TYPE WHERE USER_NAME = '${currentUser}'`,
```

Comment:
make sure you're also using ZONE_NAME here, could have bo#tempZone and bo#otherZone